File: src/test/java/com/matamercer/microblog/unit/service/PostServiceTest.java
package com.matamercer.microblog.unit.service;


import com.matamercer.microblog.models.entities.*;
import com.matamercer.microblog.models.repositories.PostRepository;
import com.matamercer.microblog.security.authorization.UserRole;
import com.matamercer.microblog.services.BlogService;
import com.matamercer.microblog.services.FileService;
import com.matamercer.microblog.services.PostService;
import com.matamercer.microblog.services.PostTagService;
import com.matamercer.microblog.web.api.v1.dto.mappers.request.PostRequestDtoMapper;
import com.matamercer.microblog.web.api.v1.dto.mappers.response.PostResponseDtoMapper;
import org.junit.Before;
import org.junit.runner.RunWith;
import org.mockito.Mockito;
import org.springframework.test.context.ActiveProfiles;
import org.springframework.test.context.junit4.SpringRunner;


import static org.assertj.core.api.Assertions.assertThat;
import static org.mockito.ArgumentMatchers.any;

@RunWith(SpringRunner.class)
@ActiveProfiles("Test")
public class PostServiceTest {

   private PostService postService;
   private Post post;
   private User user;

   @Before
   public void setup(){

      post = new Post(Mockito.mock(Blog.class),
              "title",
              "content",
              false,
              false, true);
      post.setId(1L);

      PostRepository postRepository = Mockito.mock(PostRepository.class);
      Mockito.when(postRepository.save(any(Post.class)))
              .thenAnswer(i -> i.getArguments()[0]);

      PostTagService postTagService = Mockito.mock(PostTagService.class);

      BlogService blogService = Mockito.mock(BlogService.class);

      FileService fileService = Mockito.mock(FileService.class);

      PostResponseDtoMapper postResponseDtoMapper = Mockito.mock(PostResponseDtoMapper.class);
      PostRequestDtoMapper postRequestDtoMapper = Mockito.mock(PostRequestDtoMapper.class);

      user = new User("username@gmail.com",
              "username",
              "password",
              UserRole.BLOGGER,
              true,
              true,
              true,
              true,
              AuthenticationProvider.LOCAL);
      postService = new PostService(postRepository,
              postTagService,
              blogService,
              fileService,
              postResponseDtoMapper,
              postRequestDtoMapper);
   }

//   @Test
//   public void whenCreatePost_returnCreatedPost(){
//      PostRequestDto postRequestDto = new PostRequestDto();
//      postRequestDto.setContent(post.getContent());
//      postRequestDto.setTitle(post.getTitle());
//      MultipartFile[] files = new MultipartFile[0];
//      Post createdPost = postService.createNewPost(postRequestDto, files, user.getActiveBlog());
//      assertThat(createdPost.getContent()).isEqualTo(postRequestDto.getContent());
//      assertThat(createdPost.getTitle()).isEqualTo(postRequestDto.getTitle());
//   }
//
//   @Test
//   public void whenConvertPost_returnResponseDto(){
//      PostRequestDto postRequestDto = new PostRequestDto();
//      postRequestDto.setContent(post.getContent());
//      postRequestDto.setTitle(post.getTitle());
//      MultipartFile[] files = new MultipartFile[0];
//      Post createdPost = postService.createNewPost(postRequestDto, files, user.getActiveBlog());
//
//      PostResponseDto postResponseDto = postService.convertEntityToDtoResponse(createdPost);
//      assertThat(postResponseDto.getContent()).isEqualTo(createdPost.getContent());
//   }


}

File: src/test/java/com/matamercer/microblog/unit/service/PostServiceTest.kt
package com.matamercer.microblog.unit.service

import com.matamercer.microblog.models.entities.AuthenticationProvider
import com.matamercer.microblog.models.entities.Blog
import com.matamercer.microblog.models.entities.Post
import com.matamercer.microblog.models.entities.User
import com.matamercer.microblog.models.repositories.PostRepository
import com.matamercer.microblog.security.authorization.UserRole
import com.matamercer.microblog.services.BlogService
import com.matamercer.microblog.services.FileService
import com.matamercer.microblog.services.PostService
import com.matamercer.microblog.services.PostTagService
import com.matamercer.microblog.web.api.v1.dto.requests.PostRequestDto
import com.matamercer.microblog.web.api.v1.dto.responses.PostResponseDto
import io.mockk.clearAllMocks
import io.mockk.every
import io.mockk.impl.annotations.MockK
import io.mockk.junit5.MockKExtension
import io.mockk.mockkClass
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.extension.ExtendWith
import org.springframework.test.context.ActiveProfiles
import org.springframework.web.multipart.MultipartFile

@ExtendWith(MockKExtension::class)
@ActiveProfiles("Test")
class PostServiceTest {
    @MockK
    private lateinit var postService: PostService

    @MockK
    private lateinit var postRepository: PostRepository

    @MockK
    private lateinit var postTagService: PostTagService

    @MockK
    private lateinit var blogService: BlogService

    @MockK
    private lateinit var fileService: FileService
    private lateinit var post: Post
    private var user: User? = null

    @BeforeEach
    fun setup() {
        clearAllMocks()
        post = Post(
            mockkClass(Blog::class),
            "title",
            "content",
            false,
            false, true
        )
        post.id = 1L
        user = User(
            "username@gmail.com",
            "username",
            "password",
            UserRole.BLOGGER,
            AuthenticationProvider.LOCAL
        )
        postService = PostService(
            postRepository,
            postTagService,
            blogService,
            fileService,
        )
    }

    @Test
    fun whenCreatePost_returnCreatedPost() {
        val postRequestDto = PostRequestDto(sensitive = true)
        postRequestDto.content = post.content
        postRequestDto.title = post.title
        val blog = Blog()
        blog.id = 1L
        post.blog = blog
        val files = emptyArray<MultipartFile>()
        every { postRepository.save(any()) } returns post
        val createdPost = postService.createNewPost(postRequestDto, files, blog)
        assertThat(createdPost.content).isEqualTo(postRequestDto.content)
        assertThat(createdPost.title).isEqualTo(postRequestDto.title)
    }
 }